src/Domain/CoordinateMaps/TimeDependent/Translation.cpp

```diff
@@ -204,6 +297,65 @@ std::array<tt::remove_cvref_wrap_t<T>, Dim> Translation<Dim>::coord_helper(
   return result;
 }
 
+template <size_t Dim>
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, Dim> Translation<Dim>::inspiral_helper(
+    const std::array<T, Dim>& source_coords, double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time,
+    const size_t function_or_deriv_index) const {
+  ASSERT(functions_of_time.count(f_of_t_name_) == 1,
+         "The function of time '" << f_of_t_name_
+                                  << "' is not one of the known functions of "
+                                     "time. The known functions of time are: "
+                                  << keys_of(functions_of_time));
+  const auto func_or_deriv_of_time =
+      gsl::at(functions_of_time.at(f_of_t_name_)->func_and_deriv(time),
+              function_or_deriv_index);
+  ASSERT(func_or_deriv_of_time.size() == Dim,
+         "The dimension of the function of time ("
+             << func_or_deriv_of_time.size()
+             << ") does not match the dimension of the translation map (" << Dim
+             << ").");
+  std::array<tt::remove_cvref_wrap_t<T>, Dim> result{};
+  // sizing the result and getting the radial function value
+  for (size_t i = 0; i < Dim; i++) {
+    gsl::at(result, i) = gsl::at(source_coords, i);
+  }
+  // test this for source_coords so that we can take out the for loop maybe
+  auto radius = magnitude(result);
+  if (function_or_deriv_index == 1) {
+    for (size_t i = 0; i < Dim; i++) {
+      gsl::at(result, i) = 0.0;
+    }
+  }
+  for (size_t k = 0; k < get_size(radius); k++) {
+    if (get_element(radius, k) <= inner_radius_) {
+      for (size_t i = 0; i < Dim; i++) {
+        get_element(gsl::at(result, i), k) += gsl::at(func_or_deriv_of_time, i);
+      }
+      ASSERT(max(magnitude(result)) <= outer_radius_,
+             "Coordinates translated outside outer radius, This should not"
+                 << " happen");
+    } else if (get_element(radius, k) > inner_radius_ &&
+               get_element(radius, k) < outer_radius_) {
+      // this factor comes from the SpEC implementation
```

Comment:
Can you write out in the code comment what the function form of w is and that it's a linear falloff? I don't think you need to mention SpEC here since a linear falloff is simple enough.